internal/auth/internal/controllers/trafficpermissions/controller_test.go

```diff
@@ -729,6 +732,91 @@ func (suite *controllerSuite) TestReconcile_TrafficPermissionsDelete_Destination
 	})
 }
 
+// 1. Create ALLOW traffic permission granting foo -> bar
+// 2. Observe reconciler write CTP for bar listing source foo
+// 3. User updates TP from step 1 to instead grand foo -> baz
+// 4. Observe reconciler update CTP for bar to list source baz
+// 5. (must) Observe reconciler update CTP for bar to default (no permissions)
+func TestController_OrphanedTrafficPermissions(t *testing.T) {
```

Comment:
This is a standalone "controller test" that `TestController*` tests below should be updated to follow.
It does not require manual `Reconcile` calls and tests the lifecycle of the controller in response to events (like resource writes in this test).

Reply:
On another note, this test will pass even without the BRC code added as part of this PR. The bimapper used by the controller [here](https://github.com/hashicorp/consul/blob/202090e5d55c93f9170890bc2303e5853d147c3c/internal/auth/internal/mappers/trafficpermissionsmapper/traffic_permissions_mapper.go#L52-L55) already handles orphaned CTPs for us.
However, we plan to phase out bimappers entirely so this test is a sanity check that the cleanups still happen. 

I have confirmed that if I comment out the loop to handle old CTPs linked above, this test will fail without the BRC code.